lib/dor/services/registration_service.rb

```diff
@@ -4,6 +4,47 @@ module Dor
   class RegistrationService
 
     class << self
+
+      # @TODO: Why isn't all this logic in, for example, Dor::Item.create? or Dor::Base.create? or Dor::Creatable.create?
+      # @TODO: these duplicate checks could be combined into 1 query
+
+      # @param [String] pid an ID to check, if desired.  If not passed (or nil), a new ID is minted
+      # @return [String] a pid you can use immidately, either freshly minted or your checked value
+      # @raise [Dor::DuplicateIdError]
+      def unduplicated_pid(pid = nil)
```

Comment:
This method name could be more clear. Maybe `check_or_mint_unique_pid` or something like that instead of `unduplicated_pid`? It's a little awkward that it's both a check function and a minting function.